client/data/site-profiler/test/use-hosting-provider-url-details.test.ts

```diff
@@ -0,0 +1,86 @@
+/**
+ * @jest-environment jsdom
+ */
+import { renderHook } from '@testing-library/react';
+import { useAnalyzeUrlQuery } from 'calypso/data/site-profiler/use-analyze-url-query';
+import { useHostingProviderQuery } from 'calypso/data/site-profiler/use-hosting-provider-query';
+import useHostingProviderName from 'calypso/site-profiler/hooks/use-hosting-provider-name';
+import { useHostingProviderUrlDetails } from '../use-hosting-provider-url-details';
+
+jest.mock( 'calypso/data/site-profiler/use-analyze-url-query' );
+jest.mock( 'calypso/data/site-profiler/use-hosting-provider-query' );
+jest.mock( 'calypso/site-profiler/hooks/use-hosting-provider-name' );
+
+( useAnalyzeUrlQuery as jest.Mock ).mockReturnValue( {
+	data: {},
+	isLoading: false,
+} );
+
+( useHostingProviderQuery as jest.Mock ).mockReturnValue( {
+	data: {
+		hosting_provider: {
+			slug: 'automattic',
+		},
+	},
+	isLoading: false,
+} );
+
+( useHostingProviderName as jest.Mock ).mockReturnValue( 'WordPress.com' );
+
+describe( 'useHostingProviderUrlDetails', () => {
+	it( 'returns correct values when the hosting provider is a8c', () => {
```

Comment:
We probably could apply the same [suggestion from @gabrielcaires](https://github.com/Automattic/wp-calypso/pull/91748#discussion_r1640131347) here, since the 3 tests are the same just changing variables.